common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -540,7 +540,11 @@ private void transferTimeoutsToBuckets() {
                 int stopIndex = (int) (ticks & mask);
 
                 HashedWheelBucket bucket = wheel[stopIndex];
-                bucket.addTimeout(timeout);
+                if (calculated < tick) {
```

Comment:
I wonder if this might not result into surprises in terms of ordering. As if you have multiple timeouts that are expired you will basically revert there execution order.